### PR TITLE
Extract `workers`

### PR DESCRIPTION
**Pull request by oeway**
Fixes https://github.com/oeway/ImJoy/issues/185

### CODE REVIEW

**Comment by MartinHjelmare** [comment]
Should we put the worker_template.py directly as the package `__init__.py`, and rename worker.py to client.py?

**Comment by MartinHjelmare** [comment]
It's nice to start from a fresh branch, so there are no old commits.

File: imjoy/workers/worker_template.py [hunk 1]
```diff
@@ -64,7 +70,7 @@ class PluginConnection:
 
     # pylint:disable=too-many-instance-attributes
 
-    def __init__(self, opt):
+    def __init__(self, opt, client=None):
```
Comment (MartinHjelmare) [inline imjoy/workers/worker_template.py]: When do we need to pass a client?

**Comment by oeway** [comment]
> Should we put the worker_template.py directly as the package `__init__.py`, and rename worker.py to client.py?
Maybe we should keep it for now because we may have other types of worker in the future. For example with pyimagej worker for running imagej script.
How about use full name `python_worker.py`, `python2_client.py` and `python3_client.py`?
Later we can have `pyimagej_worker.py` for example.

File: imjoy/workers/worker_template.py (hunk 1 above)
Comment (oeway) [reply imjoy/workers/worker_template.py]: This will be used in the tests where the `fake_client.test_client` will be passed in.

**Comment by oeway** [comment]
> The test module have changed access mode.

What do you mean by that?

**Comment by MartinHjelmare** [comment]
Go to https://github.com/oeway/ImJoy-Engine/pull/66/files and scroll down to the test modules. You can see that it says that access mode has changed from 644 to 755 for the test modules.

**Comment by oeway** [comment]
> Go to https://github.com/oeway/ImJoy-Engine/pull/66/files and scroll down to the test modules. You can see that it says that access mode has changed from 644 to 755 for the test modules.

I see, how should I fix it? I just tried `chmod 644 -R tests` but I then cannot run `git add tests` because `Permission denied`.

**Comment by MartinHjelmare** [comment]
Sounds weird. You shouldn't need higher than 644.

**Comment by oeway** [comment]
> Sounds weird. You shouldn't need higher than 644.

Fixed by `git checkout master -- tests/*.py`

**Comment by oeway** [comment]
I am going to merge this PR, please check if you have any other suggestions.

**Pull request merged**